packages/react/passport/src/PassportContext.tsx

```diff
@@ -0,0 +1,187 @@
+  const [isLoggedIn, setLoggedIn] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+  const [withWallet, setWithWallet] = useState(true);
+  const passportInstance = new Passport(config);
+
+  const checkLogggedIn = useCallback(async () => {
+    const p = await passportInstance.login({
+      useCachedSession: true,
+    });
+    setLoggedIn(!!p);
+  }, [passportInstance]);
+
+  useEffect(() => {
+    checkLogggedIn();
+  }, [config, passportInstance]);
+
+  const v = useMemo(() => ({
+    passportInstance,
+    isLoggedIn,
+    isLoading,
+    error,
+    withWallet,
+    login: async (options?: {
+      useCachedSession?: boolean;
+      anonymousId?: string;
+      withoutWallet?: boolean;
+    }) => {
+      try {
+        setError(null);
+        setIsLoading(true);
+        setWithWallet(!options?.withoutWallet);
+        await passportInstance.login({
+          useCachedSession: !!options?.useCachedSession,
+          anonymousId: options?.anonymousId,
+        });
+
+        if (!options?.withoutWallet) {
+          const provider = passportInstance.connectEvm();
+          await provider.request({ method: 'eth_requestAccounts' });
+        }
+        setLoggedIn(true);
+      } catch (e) {
+        setError(e as Error);
+      } finally {
+        setIsLoading(false);
+      }
+    },
+    logout: async () => {
+      try {
+        setError(null);
+        setIsLoading(true);
+        await passportInstance.logout();
+        setWithWallet(false);
+        setLoggedIn(false);
+      } catch (e) {
+        setError(e as Error);
+      } finally {
+        setIsLoading(false);
+      }
+    },
+    loginCallback: async () => {
+      await passportInstance.loginCallback();
+    },
+    linkExternalWallet: passportInstance.linkExternalWallet.bind(passportInstance),
+    getIdToken: async () => {
+      setIsLoading(true);
+      try {
+        const idToken = await passportInstance.getIdToken();
+        return idToken;
+      } catch (e: any) {
+        setError(e as Error);
+        setLoggedIn(false);
+      } finally {
+        setIsLoading(false);
+      }
+      return undefined;
+    },
+    getAccessToken: async () => {
+      setIsLoading(true);
+      try {
+        const accessToken = await passportInstance.getAccessToken();
+        return accessToken;
+      } catch (e: any) {
+        setError(e as Error);
+        setLoggedIn(false);
+      } finally {
+        setIsLoading(false);
+      }
+      return undefined;
+    },
+    getAccounts: async () => {
+      setIsLoading(true);
+      try {
+        const provider = passportInstance.connectEvm();
+        const accounts = await provider.request({ method: 'eth_requestAccounts' });
+        return accounts;
+      } catch (e: any) {
+        setError(e as Error);
+      } finally {
+        setIsLoading(false);
+      }
+      return [];
+    },
```

Comment:
`eth_requestAccounts` initialises the wallet, so if we're going to keep the `withWallet` property, then this should also call `setWithWallet`